app/src/main/java/com/github/campus_capture/bootcamp/firebase/FirebaseBackend.java

```diff
@@ -0,0 +1,159 @@
+package com.github.campus_capture.bootcamp.firebase;
+
+import androidx.annotation.NonNull;
+import androidx.test.core.app.ApplicationProvider;
+
+import com.github.campus_capture.bootcamp.AppContext;
+import com.github.campus_capture.bootcamp.authentication.Section;
+import com.github.campus_capture.bootcamp.authentication.User;
+import com.github.campus_capture.bootcamp.scoreboard.ScoreItem;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ServerValue;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class FirebaseBackend implements BackendInterface{
+    @Override
+    public CompletableFuture<Boolean> voteZone(String uid, Section s, String zonename) {
+
+        AppContext context = (AppContext) ApplicationProvider.getApplicationContext();
+        FirebaseDatabase db = context.getFirebaseDB();
+
+        //register that player uid voted
```

Comment:
The way this is built could mean that the vote may go through but the player isn't flagged as having voted... I guess this is insignificant enough that it pretty much can't happen

Reply:
yeah I should try to change that